### PR TITLE
Update the forms guide

### PR DESCRIPTION
**Pull request by matthewp**
This updates the form guide and all related demos.

Part of #5119

### CODE REVIEW

File: demos/forms/advanced-manual-validation.html [hunk 1]
```diff
@@ -11,42 +11,43 @@
...
+    static view = `
+        <form>
+            {{# if(this.phoneNumberError) }}
+                <p class="error">{{ this.phoneNumberError }}</p>
+            {{/ if }}
+
+            <p>
+                Phone Number:
+                <input type="text" value:bind="this.userPhoneNumber"
+									{{# if(this.phoneNumberError) }}class="error"{{/ if }}>
```
Comment (justinbmeyer) [inline demos/forms/advanced-manual-validation.html]: is this indentation right?

File: demos/forms/advanced-not-sticky.html [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
+import { StacheElement } from "can";
 
-Component.extend({
-	tag: "not-sticky-uppercase",
+class NotStickyUppercase extends StacheElement {
+    static view = `
+        <label>
+            scopeProp:
+            <input type="text" value:from="this.scopeProp"
+							on:change:value:to="this.scopeProp">
```
Comment (matthewp) [inline demos/forms/advanced-not-sticky.html]: Indentation

File: demos/forms/elements-select-string-to-any.html [hunk 3]
```diff
@@ -1,34 +1,34 @@
...
-		 },
-		get primitiveValueType() {
-			return typeof this.primitiveValue;
-		},
-	}
-});
+    static props = {
+        primitiveValue: {
+            default: true,
+						type: type.Any
```
Comment (matthewp) [inline demos/forms/elements-select-string-to-any.html]: Indentation

File: demos/forms/advanced-manual-validation.html (hunk 1 above)
Comment (matthewp) [reply demos/forms/advanced-manual-validation.html]: Fixed, thanks.

**Comment by matthewp** [comment]
I believe this is the cause of the tests failing, in firefox 69. Looking into a workaround: https://bugzilla.mozilla.org/show_bug.cgi?id=1561911

**Comment by phillipskevin** [comment]
@matthewp is this ready to go then? I'd like to publish the site and take a look at the guide.

**Comment by matthewp** [comment]
It is yeah, took a while to figure out CI.

**Pull request merged**